homu/quay1.py

```diff
@@ -0,0 +1,325 @@
+# API wrappers for quay
+import re
+import functools
+from contextlib import contextmanager
+
+import requests
+
+from . import utils
+from .utils import lazy_debug
+
+QUAY_SCHEME = 'https'
+QUAY_HOSTNAME = 'quay.io'
+QUAY_API = 'api/v1/'
+CSRF_TOKEN_RE = re.compile(r'__token\s?=\s?\'(.*)\';')
+
+EVENT_BUILD_SUCCESS = 'build_success'
+EVENT_BUILD_FAILURE = 'build_failure'
+
+# import http.client as http_client
+# http_client.HTTPConnection.debuglevel = 1
+
+class QuayThing:
```

Comment:
%) no, that's like `Entity` or `Object`. `github3` uses `GitHubObject` as name for base class that `Repo`, `Hook`, etc extends. We can use that to or stick with "The Thing" %)